modin/experimental/backends/omnisci/query_compiler.py

```diff
@@ -293,20 +294,70 @@ def min(self, **kwargs):
 
     def sum(self, **kwargs):
+        min_count = kwargs.pop("min_count")
+        if min_count != 0:
+            return super().sum(
+                min_count=min_count,
+                **kwargs,
+            )
         return self._agg("sum", **kwargs)
 
     def mean(self, **kwargs):
         return self._agg("mean", **kwargs)
 
+    def nunique(self, axis=0, dropna=True):
+        if axis != 0 or dropna is False:
+            return super().nunique(axis=axis, dropna=dropna)
+        # TODO: check that the OmniSci server is running with the enabled watchdog
+        if "float" in self.dtypes.values:  # and is_watchdog_enabled():
+            ErrorMessage.single_warning(
+                "'DataFrame.nunique' can't be performed fast enough with the OmniSci backend "
+                "because of the columns with the 'float' type, operation will be defaulted to pandas. "
+                "To use OmniSci implementation get rid of 'float' columns (you may drop those or "
+                "round its values to integer)."
+            )
+            return super().nunique(axis=axis, dropna=dropna)
+
+        return self._agg("count", distinct=True)
+
     def _agg(self, agg, axis=0, level=None, **kwargs):
+        """
+        Perform specified aggregation along rows/columns.
+
+        Parameters
+        ----------
+        agg : str
+            Name of the aggregation function to perform.
+        axis : {0, 1}, default: 0
+            Axis to perform aggregation alogn. 0 is for columns when 1 is for rows.
+            *Note:* OmniSci backend supports aggregation for 0 axis only, aggregation
+            along rows will be defaulted to pandas.
+        level : None
+            Serves the compatibility purpose, always have to be None.
+        **kwargs : dict
+            Additional parameters to pass to the aggregation function.
+
+        Returns
+        -------
+        DFAlgQueryCompiler
+            New single-column (``axis=1``) or single-row (``axis=0``) query compiler containing
+            the result of aggregation.
+        """
         if level is not None or axis != 0:
             return getattr(super(), agg)(axis=axis, level=level, **kwargs)
 
-        skipna = kwargs.get("skipna", True)
-        if skipna is False:
+        # TODO: Do filtering on numeric columns if `numeric_only=True`
+        if kwargs.get("skipna", True) is False or kwargs.get("numeric_only") is True:
             return getattr(super(), agg)(axis=axis, level=level, **kwargs)
-
-        new_frame = self._modin_frame.agg(agg)
+        # Processed above, so can be omitted
+        kwargs.pop("skipna", None)
+        kwargs.pop("numeric_only", None)
+        # `squeeze_self` parameter was added to the base query compiler as a workaround
+        # for pandas-issue#41074 which is specific for pandas backends only, this parameter
+        # has no effect for OmniSci backend and so can be omitted.
+        kwargs.pop("squeeze_self", None)
```

Comment:
This should be removed after #3161 merge.

Reply:
removed